packages/symbiosis/src/Symbiosis.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  type TransactionFilter,
+  type BridgeActionParams,
+  compressJson,
+} from '@rabbitholegg/questdk'
+import { type Address } from 'viem'
+import { CHAIN_ID_ARRAY, CHAIN_TO_TOKENS } from './constants'
+import { metaBurnABI, metaRouteABI, metaSynthesizeABI } from './abi'
+import { symbiosis } from './symbiosis-sdk'
+
+export const bridge = async (
+  bridge: BridgeActionParams,
+): Promise<TransactionFilter> => {
+  const {
+    sourceChainId,
+    destinationChainId,
+    contractAddress,
+    tokenAddress,
+    amount,
+    recipient,
+  } = bridge
+
+  const bridgeContract: Address =
+    contractAddress ?? (symbiosis.metaRouter(sourceChainId).address as Address)
+
+  return compressJson({
+    chainId: sourceChainId,
+    to: bridgeContract,
+    input: {
+      $abi: metaRouteABI,
+      _metarouteTransaction: {
+        approvedTokens: tokenAddress ? [tokenAddress] : undefined, // if tokenAddress is undefined, any input token will pass filter
+        amount: amount,
+        otherSideCalldata: {
```

Comment:
Right here you should be able to do 
```
otherSideCalldata: {
    $abiAbstract: metaBurnABI,
    _metaBurnTransaction: {
        chainID: destinationChainId,
        chain2address: recipient,
    }
}
```

And that should work - I wouldn't bother cause this looks like it's working fine and it's more precise but just to see how you'd use that.

Reply:
I like this solution much better, and it allows us to remove the abi for the metaSynthesize function.

Im going to go ahead and implement this.